pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -908,3 +994,7 @@ func buildHostnameMatch(hostname *k8s.Hostname) []string {
 func emptyOrEqual(have, expected string) bool {
 	return have == "" || have == expected
 }
+
+func StrPointer(s string) *string {
```

Comment:
Any idea why their APIs are using string pointers? Golang is pretty optimized for strings ... just curious the motivation.

Reply:
I don't really understand the motivation behind the motivation, but something about optional fields are supposed to be pointers by k8s conventions? Not sure why if they are not distinguishing between `""` and `nil` anywhere though